extensions/positron-python/src/client/positron/interpreterSettings.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import path from 'path';
+
+import { traceInfo, traceVerbose } from '../logging';
+import { getConfiguration } from '../common/vscodeApis/workspaceApis';
+import { arePathsSame, isParentPath } from '../pythonEnvironments/common/externalDependencies';
+import { INTERPRETERS_EXCLUDE_SETTING_KEY, INTERPRETERS_INCLUDE_SETTING_KEY } from '../common/constants';
+
+/**
+ * Gets the list of interpreters that the user has explicitly included in the settings. Converts
+ * relative and aliased paths to absolute paths.
+ * @returns List of interpreters that the user has explicitly included in the settings.
+ */
+export function getUserIncludedInterpreters(): string[] {
+    const interpretersInclude = getConfiguration('python').get<string[]>(INTERPRETERS_INCLUDE_SETTING_KEY) ?? [];
+    if (interpretersInclude.length > 0) {
+        return interpretersInclude.filter((item) => {
+            if (path.isAbsolute(item)) {
+                return true;
+            }
+            traceVerbose(`[shouldIncludeInterpreter]: included interpreter path ${item} is not absolute...ignoring`);
+            return false;
+        });
+    }
+    traceVerbose(`[shouldIncludeInterpreter]: No interpreters specified via ${INTERPRETERS_INCLUDE_SETTING_KEY}`);
+    return [];
+}
+
+/**
+ * Gets the list of interpreters that the user has explicitly excluded in the settings. Converts
+ * relative and aliased paths to absolute paths.
+ * @returns List of interpreters that the user has explicitly excluded in the settings.
+ */
+export function getUserExcludedInterpreters(): string[] {
+    const interpretersExclude = getConfiguration('python').get<string[]>(INTERPRETERS_EXCLUDE_SETTING_KEY) ?? [];
+    if (interpretersExclude.length > 0) {
+        return interpretersExclude.filter((item) => {
+            if (path.isAbsolute(item)) {
+                return true;
+            }
+            traceVerbose(`[shouldIncludeInterpreter]: excluded interpreter path ${item} is not absolute...ignoring`);
+            return false;
+        });
+    }
+    traceVerbose(`[shouldIncludeInterpreter]: No interpreters specified via ${INTERPRETERS_EXCLUDE_SETTING_KEY}`);
+    return [];
+}
+
+/**
+ * Check whether an interpreter should be included in the list of discovered interpreters.
+ * If an interpreter is both explicitly included and excluded, it will be included.
```

Comment:
Are you confident this is the right approach? I'm wondering if we'd run into a situation where a Workbench admin has explicitly excluded an interpreter, but then a confused user tries to include it. We might want to consider implementing a way to log this use for admins, but we don't have monitoring implemented yet in Workbench for VS Code or Positron.

Reply:
The language pack output will have a bit of info about why certain interpreters were included/excluded, but longer term, I'm hoping to better surface this information in a dedicated pane or something similar.

I think this approach makes sense for the Desktop user case, but let me think on the admin situation a bit! I feel like we will want a separate flag/option elsewhere, like an "admin policy" to enforce certain options